retrieval/scrape.go

```diff
@@ -446,6 +461,28 @@ func (sl *scrapeLoop) stop() {
 	<-sl.done
 }
 
+func (sl *scrapeLoop) processScrapeResult(samples model.Samples, scrapeErr error, start time.Time) error {
+	// Collect samples post-relabelling and label handling in a buffer.
+	buf := &bufferAppender{buffer: make(model.Samples, 0, len(samples))}
```

Comment:
It was a 50k samples/s setup, and I looked at total allocated space.

I can't tell you anything about heap fragmentation, as I estimate this setup would have taken at least a week to stabilise memory wise due to chunkdescs.

> In different news, if 5 lines of code save 0.5% of allocation, I would go for it.

You'll love my other PR then. It removes 6 lines net for a 17% reduction.